WordPress/openverse
# Fix frontend robots.txt

## Fixes



Fixes a regression introduced in #4077 

## Description



This PR returns to using a sever middleware for robots.txt in the Openverse frontend. It also adds robots.txt to the frontend gitignore to help prevent us from accidentally adding a static robots.txt file in the future.

It also adds Crawl-delay to slow down bing and other search engines when they crawl the pages we _do_ allow them to crawl.

## Testing Instructions


Run the frontend locally with `just frontend/run dev` and verify that you see the correct file contents as defined in [/frontend/src/server-middleware/robots.js](https://github.com/WordPress/openverse/blob/fix-frontend-robots/frontend/src/server-middleware/robots.js#L65)

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
